Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -463,7 +469,15 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
    */
   virtual TYPE &operator=(const TYPE &value) {
     TYPE oldValue = m_value;
-    m_value = value;
+    if (std::is_same<TYPE, std::string>::value) {
```

Comment:
I'm wondering if one can write `myProperty = "StringValue";` instead of `myProperty = std::string("StringValue");`. It would be nice to see both in the unit tests.